### ISSUE TITLE
Need a /dev/dsp work around for ASL. /dev/dsp was dropped from Debian around v10

### ISSUE THREAD

**Issue by N6GKJ** [Allan-N/ASL-Install#3]
There is no /dev/dsp in the Ubuntu distribution so we need a way to get /dev/snd to work. Since Asterisk needs to have /dev/dsp in order to use Sound FOBS such as URI or RIM we need to figure this out

**Comment by Allan-N** [comment]
Can you check your "/etc/modules" file to see if it includes "snd_pcm_oss".  If not, please add that line, reboot, and check to see if "/dev/dsp" is now present.  ... and, if so, did that get your configuration up and running?

**Comment by N6GKJ** [comment]
Can you sent the proper text format as to what the line should look like?

**Comment by Allan-N** [comment]
If missing from the file then you should (as root) be able to use :

`echo snd_pcm_oss >> /etc/modules`

**Comment by N6GKJ** [comment]
Still no /dev/dsp . But at least now I can see that /etc/modules now includes snd_pcm_oss

**Comment by N6GKJ** [comment]
6gkj@n6gkj-HP-ZBook-15-G6:/etc$ sudo insmod snd_pcm_oss
[sudo] password for n6gkj:           
insmod: ERROR: could not load module snd_pcm_oss: No such file or directory

Am I missing sosmething? I can see /dev/snd directory

**Comment by Allan-N** [comment]
OK.  Well, I had one report that adding that line to the /etc/modules file solved the problem on a Debian install.

What OS are you running?  and on what type of device?

**Comment by N6GKJ** [comment]
I am Running Linux Mint 21.1 Mate. Device is a HP-Zbook-15-G6, i7 core processor 32GB Ram

**Comment by Allan-N** [comment]
> 6gkj@n6gkj-HP-ZBook-15-G6:/etc$ sudo insmod snd_pcm_oss [sudo] password for n6gkj: insmod: ERROR: could not load module snd_pcm_oss: No such file or directory
> 
> Am I missing sosmething? I can see /dev/snd directory

Here, I would guess that your install is missing "a" package.  Not sure which one you need (or if it's even available for your distro)

**Comment by N6GKJ** [comment]
I have Intel, NIVIDIA and Realtek drivers installed in the WIN11 Drive on the same computer. I will look at dmsg and find out what the hardware is reporting. Maybe need a driver update

**Comment by N6GKJ** [comment]
but... sound works fine on the Linux Mint. I was watching YouTube earliuer and had good audio

**Comment by N6GKJ** [comment]
I can put Ubuntu Jammy on it to see if the same resut occurs, what do you think?

**Comment by Allan-N** [comment]
So far, my focus has been on cloud VM's w/no USB devices.  At this time, I can't personally confirm that any systemOS will work with the USB devices but I got one "All good now" for a Debian install.  I've got the pieces to build a local node with a CM108 but they are in the stack of hardware projects that are competing for my time.  Once I've got some hardware then I can start to experiment and see what will move things forward.

As to the question of trying Ubuntu ... I'd probably start with Debian.  Then, I'd go to Ubuntu.
